Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2004-11-28
Initial Package Version: 4.0.6
Origin: http://www.geocities.com/robm351/uclibc/shadow-uclibc.diff.txt
Upstream Status: Not submitted
Description: Shadow needs this patch to build with uClibc. If we do not
have PAM then NIS is used, but uClibc does not have NIS, so a refference
to innetgr() causes the build to fail.

Also see:
http://www.uclibc.org/
http://www.linuxfromscratch.org/hlfs/

diff -Naur shadow-4.0.6.orig/libmisc/login_access.c shadow-4.0.6/libmisc/login_access.c
--- shadow-4.0.6.orig/libmisc/login_access.c	2004-11-28 16:45:36.897206192 +0000
+++ shadow-4.0.6/libmisc/login_access.c	2004-11-28 16:48:46.474386080 +0000
@@ -192,6 +192,7 @@
 static int
 netgroup_match (const char *group, const char *machine, const char *user)
 {
+#ifdef NIS
 	static char *mydomain = 0;
 
 	if (mydomain == 0) {
@@ -202,6 +203,10 @@
 	}
 
 	return innetgr (group, machine, user, mydomain);
+#else
+	syslog(LOG_ERR, "NIS netgroup support not configured");
+	return (NO);
+#endif
 }
 
 /* user_match - match a username against one token */
